sdk/identity/identity/src/credentials/managedIdentityCredential.ts

```diff
@@ -1,34 +1,204 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import { AccessToken, GetTokenOptions, TokenCredential } from "@azure/core-http";
+import qs from "qs";
+import {
+  AccessToken,
+  GetTokenOptions,
+  RequestPrepareOptions,
+  RestError,
+  TokenCredential
+} from "@azure/core-http";
 import { IdentityClientOptions, IdentityClient } from "../client/identityClient";
 
+const DefaultScopeSuffix = "/.default";
+export const ImdsEndpoint = "http://169.254.169.254/metadata/identity/oauth2/token";
```

Comment:
Oh wow! Is this a fixed IP? Shouldn't we use a domain instead?

Reply:
This is the standard address for the IMDS endpoint as dictated by Azure VM's managed service identity implementation, they don't use a domain for it since it's hosted the local subnet of the VM itself.